infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultAdapterForUpgrade.java

```diff
@@ -81,6 +81,8 @@ public String getVersion() {
                         ((LogReplicationStreams.Version)txn.getRecord(VERSION_TEST_TABLE,
                                 versionStringKey).getPayload()).getVersion();
                 txn.commit();
+            } catch (Exception e) {
+                // Just for wrap this up
```

Comment:
The try clause here has already applied the `try-with-resource` pattern. The exception clause here is for that, in this testing purpose adapter we introduced a `TEST_VERSION_TABLE` to emulate version change, so when the version key wasn't explicitly set we want to return the default value instead of throw the exception here.

Reply:
Sorry, I did not get your point..I understand the catch block is for the try-with-resource on line 79..but since its empty, why not completely remove it?

Reply:
Because in this adapter, we want to return "version_latest" by default (i.e. if VERSION_TEST_TABLE not explicit set). So if we don't have this catch block, NPE/NoSuchElementException will be thrown instead of returning the default version string.